xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp

```diff
@@ -233,7 +233,7 @@ bool CActiveAEBufferPoolResample::Create(unsigned int totaltime, bool remap, boo
       if (m_processor->GetChannelLayout().Count() > 2)                    /* Disable upmix for CActiveAEResample if DSP layout > 2.0, becomes perfomed by DSP */
         upmix = false;
 
-      m_dspBuffer = new CActiveAEBufferPool(m_inputFormat);               /* Get dsp processing buffer class, based on dsp output format */
+      m_dspBuffer = new CActiveAEBufferPool(m_format);               /* Get dsp processing buffer class, based on dsp output format */
```

Comment:
> m_inputFormat -> ADSP -> m_adspFormat -> resampler/remapper -> m_format

Correct. AudioDSP also changed the output format from the buffer, which also produces issues.

That's what I learned from my journey and is already implemented in my [fork](https://github.com/AchimTuran/xbmc/blob/AudioDSP/V2/usability_core_fixes/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp#L300-L326). Currently it does not work with multi channel layouts above 2.0. But I guess I will also find a solution for this issue. With 2.0 AudioDSP runs very smooth now :-)